cmd/workspace/workspace/export_dir.go

```diff
@@ -59,18 +64,29 @@ func (opts exportDirOptions) callback(ctx context.Context, workspaceFiler filer.
 			return cmdio.RenderWithTemplate(ctx, newFileSkippedEvent(relPath, targetPath), "", "{{.SourcePath}} -> {{.TargetPath}} (skipped; already exists)\n")
 		}
 
-		// create the file
-		f, err := os.Create(targetPath)
+		// Write content to the local file
+		r, err := workspaceFiler.Read(ctx, relPath)
 		if err != nil {
+			// Check if this is a file size limit error
+			var aerr *apierr.APIError
+			if errors.As(err, &aerr) && aerr.StatusCode == http.StatusBadRequest && strings.Contains(aerr.Message, "exceeded max size") {
```

Comment:
APIError contains a method `ErrorDetails`, maybe it has information on the type of error instead of relying on the message parsing?

https://github.com/databricks/databricks-sdk-go/blob/56b24c3b3b8658f684b95b92e0343b5daa894636/apierr/details.go#L10